src/NuGet.Clients/NuGet.CommandLine/Commands/SearchCommand.cs

```diff
@@ -0,0 +1,178 @@
+    public class SearchCommand : Command
+    {
+        private readonly List<string> _sources = new List<string>();
+
+        [Option(typeof(NuGetCommand), "SearchCommandSourceDescription")]
+        public ICollection<string> Source
+        {
+            get { return _sources; }
+        }
+
+        [Option(typeof(NuGetCommand), "SearchCommandPreRelease")]
+        public bool PreRelease { get; set; } = false;
+
+        [Option(typeof(NuGetCommand), "SearchCommandTake")]
+        public int Take { get; set; } = 20;
+
+        private IList<PackageSource> GetEndpointsAsync()
+        {
+            List<PackageSource> configurationSources = SourceProvider.LoadPackageSources()
+                .Where(p => p.IsEnabled)
+                .ToList();
+
+            IList<PackageSource> packageSources;
+            if (Source.Count > 0)
+            {
+                packageSources = Source
+                    .Select(s => PackageSourceProviderExtensions.ResolveSource(configurationSources, s))
+                    .ToList();
+            }
+            else
+            {
+                packageSources = configurationSources;
+            }
+            return packageSources;
+        }
+
+        public override async Task ExecuteCommandAsync()
+        {
+            int lineSeparatorLength = 20;
+            string sourceSeparator = new string('=', lineSeparatorLength);
+            string packageSeparator = new string('-', lineSeparatorLength);
+
+            ILogger logger = Console;
+            CancellationToken cancellationToken = CancellationToken.None;
+
+            SearchFilter searchFilter = new SearchFilter(includePrerelease: PreRelease);
+
+            IList<PackageSource> listEndpoints = GetEndpointsAsync();
+
+            foreach (PackageSource source in listEndpoints)
+            {
+                SourceRepository repository = Repository.Factory.GetCoreV3(source);
+                PackageSearchResource resource = await repository.GetResourceAsync<PackageSearchResource>();
+
+                if (resource is null)
+                {
+                    continue;
+                }
+
+                IEnumerable<IPackageSearchMetadata> results = await resource.SearchAsync(
+                    string.Join(" ", Arguments).Trim(), // The arguments are joined with spaces to form a single query string
+                    searchFilter,
+                    skip: 0,
+                    take: Take,
+                    logger,
+                    cancellationToken);
+
+                Console.WriteLine(sourceSeparator);
+                Console.WriteLine($"Source: {source.Name}");
+
+                if (!results.Any())
+                {
+                    Console.WriteLine(packageSeparator);
+                    Console.WriteLine("No results found.");
+                    Console.WriteLine(packageSeparator + "\n");
```

Comment:
Maybe better to use `Environment.NewLine` instead of "\n".

Reply:
Also I usually see no result logic on goes 2nd and with result one goes first. Maybe just preference.

Reply:
Are you talking about which order the sources are shown in when multiple sources are queried? If yes, then the sources are taken in the order that they're found in in the NuGet.config file.

Reply:
`if(result.Any())
{
 ... 
}
else
{
...
}`